### PR TITLE
Add DNS helper module to resolve names with NAPTR record

### PR DESCRIPTION
**Pull request by surik**

### CODE REVIEW

File: src/ergw_dns.erl [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    case application:get_env(ergw, nameservers, []) of
+        [] -> inet_res:resolve(Name, in, naptr);
+        Nameservers ->
+            inet_res:resolve(Name, in, naptr, [{nameservers, Nameservers}])
+    end.
+
+-spec lookup(Name :: string() | inet:ip_address()) -> 
+    list(inet:ip_address()).
+lookup(Name) -> 
+    Response = ?MODULE:naptr(Name),
```
Comment (RoadRunnr) [inline src/ergw_dns.erl]: is there a reason for the explicit ?MODULE refrence?

Comment (surik) [reply src/ergw_dns.erl]: to work with `meck` in tests

File: src/ergw_dns.erl [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+%%====================================================================
+%% API
+%%====================================================================
+
+%% @private
+naptr(Name) ->
+    case application:get_env(ergw, nameservers, []) of
+        [] -> inet_res:resolve(Name, in, naptr);
+        Nameservers ->
+            inet_res:resolve(Name, in, naptr, [{nameservers, Nameservers}])
```
Comment (RoadRunnr) [inline src/ergw_dns.erl]: @mgumz I would think that we can rely on the DNS config of the host/vm/pod. Do we really need to make the name server configurable?

File: src/ergw_dns.erl [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+-module(ergw_dns).
+
+-export([naptr/1, lookup/1, lookup/2]).
+
+%%====================================================================
+%% API
+%%====================================================================
+
+%% @private
+naptr(Name) ->
```
Comment (RoadRunnr) [inline src/ergw_dns.erl]: Sorry, for not noticing earlier.

Erlang DNS will by default use UDP and not retry with TCP if the answer packet gets truncated.
So please, change that to
```
naptr(Name) ->
    NsOpts =
	case setup:get_env(ergw, nameservers) of
	    undefined ->
		[];
	{ok, Nameservers} ->
		[{nameservers, Nameservers}]
	end,
    %% 3GPP DNS answers are almost always large, use TCP by default....
    inet_res:resolve(Name, in, naptr, [{usevc, true} | NsOpts]).
```

Comment (surik) [reply src/ergw_dns.erl]: Changed

**Pull request merged**